packages/eas-cli/src/utils/fingerprintCli.ts

```diff
@@ -20,17 +23,40 @@ export async function createFingerprintAsync(
   if (!fingerprintPath) {
     return null;
   }
-  const Fingerprint = require(fingerprintPath);
-  const fingerprintOptions: Record<string, any> = {};
-  if (options.platform) {
-    fingerprintOptions.platforms = [options.platform];
-  }
-  if (options.workflow === Workflow.MANAGED) {
-    fingerprintOptions.ignorePaths = ['android/**/*', 'ios/**/*'];
+
+  if (process.env.EAS_SKIP_AUTO_FINGERPRINT) {
+    Log.log('Skipping project fingerprint');
+    return null;
   }
-  if (options.debug) {
-    fingerprintOptions.debug = true;
+
+  const timeoutId = setTimeout(() => {
+    Log.log('⌛️ Computing project fingerprint taking longer than expected...');
+    Log.log('⏩ To skip this step, set the environment variable: EAS_SKIP_AUTO_FINGERPRINT=1');
+  }, 5000);
+
+  const spinner = ora(`Computing project fingerprint`).start();
+  try {
+    const Fingerprint = require(fingerprintPath);
+    const fingerprintOptions: Record<string, any> = {};
+    if (options.platform) {
+      fingerprintOptions.platforms = [options.platform];
+    }
+    if (options.workflow === Workflow.MANAGED) {
+      fingerprintOptions.ignorePaths = ['android/**/*', 'ios/**/*'];
+    }
+    if (options.debug) {
+      fingerprintOptions.debug = true;
+    }
+    const fingerprint = await Fingerprint.createFingerprintAsync(projectDir, fingerprintOptions);
```

Comment:
I forgot this when I was reviewing the other PR re: doing a function call directly vs CLI call. One additional consideration/benefit of CLI is the process/subprocess model of using the CLI via spawn instead of a direct function call. By having this be in a subprocess rather than the CLI process, we could monitor that and even kill it with a `Promise.race` if it's taking too long (I think?).